src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -759,238 +690,17 @@ public void setupOnline()
             setCrcCheckChance(cfs.getCrcCheckChance());
     }
 
-    /**
-     * See {@link #load(boolean, boolean)}
-     * @param validation Metadata for SSTable being loaded
-     * @param isOffline Whether the SSTable is being loaded by an offline tool (sstabledump, scrub, etc)
-     * @throws IOException
-     */
-    private void load(ValidationMetadata validation, boolean isOffline) throws IOException
-    {
-        if (metadata().params.bloomFilterFpChance == 1.0)
-        {
-            // bf is disabled.
-            load(false, !isOffline);
-            bf = FilterFactory.AlwaysPresent;
-        }
-        else if (!components.contains(Component.PRIMARY_INDEX)) // What happens if filter component and primary index is missing?
-        {
-            // avoid any reading of the missing primary index component.
-            // this should only happen during StandaloneScrubber
-            load(false, !isOffline);
-        }
-        else if (!components.contains(Component.FILTER) || validation == null)
-        {
-            // bf is enabled, but filter component is missing.
-            load(!isOffline, !isOffline);
-            if (isOffline)
-                bf = FilterFactory.AlwaysPresent;
-        }
-        else
-        {
-            // bf is enabled and fp chance matches the currently configured value.
-            load(false, !isOffline);
-            loadBloomFilter(descriptor.version.hasOldBfFormat());
-        }
-    }
-
     /**
      * Load bloom filter from Filter.db file.
      *
      * @throws IOException
-     * @param oldBfFormat
      */
-    private void loadBloomFilter(boolean oldBfFormat) throws IOException
+    private static IFilter loadBloomFilter(Descriptor descriptor) throws IOException
```

Comment:
`loadBloomFilter()` might be a good candidate to just inline, or at least move to the builder.

Reply:
+1 moved for `Builder`